torch/testing/_creation.py

```diff
@@ -126,7 +126,9 @@ def clamp(a, l, h):
         rand_val = torch.rand(shape, device=device, dtype=dtype)
         result = high * rand_val + low * (1 - rand_val)
     elif dtype in _complex_types:
-        float_dtype = torch.float if dtype is torch.cfloat else torch.double
+        float_dtype = {torch.complex32: torch.float16,
```

Comment:
IIRC, this same logic was also being used in an earlier PR. might make sense to add a helper function?